app/uk/gov/hmrc/fileupload/read/envelope/model.scala

```diff
@@ -66,6 +73,18 @@ object Envelope {
     val rawData = json.asInstanceOf[JsObject] ++ Json.obj("_id" -> _id)
     Json.fromJson[Envelope](rawData).get
   }
+
```

Comment:
I don't like this method. What does `loadConstraints` mean? Loading from where? Names should reflect what code does. Here it provides defaults if they were not provided by the user. So how about a name such as `constraintsWithDefaultsIfNotProvided`? Long? Fair enough. So maybe sth shorter but still descriptive?

Also this method is very complex and **should be unit tested on its own**, it would be very easy to introduce a bug here. In fact there seems to be a bug => what is going to happen if `maxNumFiles` and `maxSizePerItem` are both provided but `maxSize` not?

Imho this method could be simplified, how about sth like this (as instance method on the req object):

```
case class CreateEnvelopeRequest(callbackUrl: Option[String] = None,
                                 expiryDate: Option[DateTime] = None,
                                 metadata: Option[JsObject] = None,
                                 constraints: Option[Constraints] = None) {
  def constraintsWithDefaultsIfNotProvided =
    Constraints(
      maxNumFiles = constraints.map(_.maxNumFiles).getOrElse(defaultMaxCapacity),
      maxSize = constraints.map(_.maxSize).getOrElse(defaultMaxSize),
      maxSizePerItem = constraints.map(_.maxSizePerItem).getOrElse(defaultMaxSizePerItem)
    )

}
```